python3/cloudwatch/src/lambda_function.py

```diff
@@ -88,11 +89,19 @@ def _parse_to_json(log):
 def _parse_cloudwatch_log(log, additional_data):
     # type: (dict, dict) -> bool
     _add_timestamp(log)
+
     if LAMBDA_LOG_GROUP in additional_data['logGroup']:
         if _is_valid_log(log):
             _extract_lambda_log_message(log)
         else:
             return False
+
+    if _has_whitelist():
```

Comment:
Why there' isn't a `_has_blacklist()`?
Also, better to have those checks (_has_whitelist, _has_blacklist) before the function, so that we won't check them again and again for each and every log, but one time.

Reply:
Rewrote the implementation a bit to make this clearer and avoid duplication